tests/functional/adapter/test_snapshot.py

```diff
@@ -0,0 +1,204 @@
+    add_column
+)
+
+
+_SEED = """
+id,first_name,last_name,email,gender,ip_address,updated_at
+1,Judith,Kennedy,jkennedy0@phpbb.com,Female,54.60.24.128,2015-12-24
+2,Arthur,Kelly,akelly1@eepurl.com,Male,62.56.24.215,2015-10-28
+3,Rachel,Moreno,rmoreno2@msu.edu,Female,31.222.249.23,2016-04-05
+4,Ralph,Turner,rturner3@hp.com,Male,157.83.76.114,2016-08-08
+5,Laura,Gonzales,lgonzales4@howstuffworks.com,Female,30.54.105.168,2016-09-01
+6,Katherine,Lopez,klopez5@yahoo.co.jp,Female,169.138.46.89,2016-08-30
+7,Jeremy,Hamilton,jhamilton6@mozilla.org,Male,231.189.13.133,2016-07-17
+8,Heather,Rose,hrose7@goodreads.com,Female,87.165.201.65,2015-12-29
+9,Gregory,Kelly,gkelly8@trellian.com,Male,154.209.99.7,2016-03-24
+10,Rachel,Lopez,rlopez9@themeforest.net,Female,237.165.82.71,2016-08-20
+11,Donna,Welch,dwelcha@shutterfly.com,Female,103.33.110.138,2016-02-27
+12,Russell,Lawrence,rlawrenceb@qq.com,Male,189.115.73.4,2016-06-11
+13,Michelle,Montgomery,mmontgomeryc@scientificamerican.com,Female,243.220.95.82,2016-06-18
+14,Walter,Castillo,wcastillod@pagesperso-orange.fr,Male,71.159.238.196,2016-10-06
+15,Robin,Mills,rmillse@vkontakte.ru,Female,172.190.5.50,2016-10-31
+16,Raymond,Holmes,rholmesf@usgs.gov,Male,148.153.166.95,2016-10-03
+17,Gary,Bishop,gbishopg@plala.or.jp,Male,161.108.182.13,2016-08-29
+18,Anna,Riley,arileyh@nasa.gov,Female,253.31.108.22,2015-12-11
+19,Sarah,Knight,sknighti@foxnews.com,Female,222.220.123.177,2016-09-26
+20,Phyllis,Fox,null,Female,163.191.232.95,2016-08-21
+21,Judy,Robinson,jrobinsonk@blogs.com,Female,208.21.192.232,2016-09-18
+22,Kevin,Alvarez,kalvarezl@buzzfeed.com,Male,228.106.146.9,2016-07-29
+23,Barbara,Carr,bcarrm@pen.io,Female,106.165.140.17,2015-09-24
+24,William,Watkins,wwatkinsn@guardian.co.uk,Male,78.155.84.6,2016-03-08
+25,Judy,Cooper,jcoopero@google.com.au,Female,24.149.123.184,2016-10-05
+26,Shirley,Castillo,scastillop@samsung.com,Female,129.252.181.12,2016-06-20
+27,Justin,Harper,jharperq@opera.com,Male,131.172.103.218,2016-05-21
+28,Marie,Medina,mmedinar@nhs.uk,Female,188.119.125.67,2015-10-08
+29,Kelly,Edwards,kedwardss@phoca.cz,Female,47.121.157.66,2015-09-15
+30,Carl,Coleman,ccolemant@wikipedia.org,Male,82.227.154.83,2016-05-26
+""".strip()
+
+
+_MODEL = """
+{{ config(materialized="table") }}
+select * from {{ ref('seed') }}
+where id between 1 and 20
+"""
+
+_SNAPSHOT_TIMESTAMP = """
+{% snapshot snapshot %}
+    {{ config(
+        target_database=database,
+        target_schema=schema,
+        unique_key='id',
+        strategy='timestamp',
+        updated_at='updated_at',
+        invalidate_hard_deletes=True,
+    ) }}
+    select * from {{ ref('fact') }}
+{% endsnapshot %}
+"""
+
+
+_SNAPSHOT_CHECK = """
+{% snapshot snapshot %}
+    {{ config(
+        target_database=database,
+        target_schema=schema,
+        unique_key='id',
+        strategy='check',
+        check_cols=['email'],
+    ) }}
+    select * from {{ ref('fact') }}
+{% endsnapshot %}
+"""
+
+
+class SnapshotBase:
```

Comment:
Looks like dbt-core has a snapshot test: https://github.com/dbt-labs/dbt-core/blob/main/tests/functional/simple_snapshot/test_basic_snapshot.py

Could we just move those tests to the adapter zone and inherit from them or if these are better move these into core?

Reply:
I don't want to inherit tests from another repo. I think it's a bad pattern.

- The foreign repo tests become too integrated with the functional code and you wind up with a polluted interface
- It leads to the scenario where you need to submit parallel PRs in multiple repos to update a test case
- It's not as easy to see what's actually being tested

Reply:
Re-evaluating our pattern for test inheritance/management is probably out of scope for just converting our tests. Not saying that the current pattern is ideal but it's also not ideal to have a one off test following it's own pattern.

Reply:
After an internal discussion with the team, we sad we'd move forward with this approach in this scenario and submit a proposal to follow this moving forward.